### ISSUE TITLE
A few requests for ephemeris visualization (AstroCalc)

### ISSUE THREAD

**Issue by Atque** [Stellarium/stellarium#1315]
Not urgent!

I wish for a few improvements of the ephemeris visualization in AstroCalc.

1. Being able to change the symbol of the marking. E.g. a cross, a star, a ring, general comet symbol etc.
2. Being able to show markings only when labled. This means that I can choose to show e.g. Hale-Bopp's position from 1997-01-01 to 1997-12-31, with 1 day intervals, but only show the markings at e.g. 10 day intervals. This means the line keeps smooth, but without markings in every breaking point. See this image for a suggestion:
https://upload.wikimedia.org/wikipedia/commons/f/fc/Comet_Hale-Bopp_starmap_1997.png
3. Being able to use more customizable labels. For example, I would like to show only day and month, or year and month, or just day and hour.
4. Perhaps some antialiasing of the lines and symbols?

**Comment by axd1967** [comment]
this would be nice for comets; if I understand well, Stellarium already has a drawing model for those.

so one would see the orientation of a comet along the predicted positions.

(I think that is big enough to be dealt with in a separate issue.)

**Comment by gzotti** [comment]
Far from trivial. A general comet icon could be drawn rotated with a tail pointing away from the sun. But the appearance of the tail model cannot be queried.

**Comment by axd1967** [comment]
> Far from trivial. A general comet icon could be drawn rotated with a tail pointing away from the sun. But the appearance of the tail model cannot be queried.

So the current representation of comets is also an approximate? Wouldn't that be a good starter?
![image](https://user-images.githubusercontent.com/3529789/95345783-6a48e200-08bb-11eb-80e1-b6281e1d56bb.png)

**Comment by gzotti** [comment]
Alright. Looks like the next comet with false MPC data. Please edit its elements and reduce absolute magnitude by ~20 or 23 magnitudes... Or did I miss something big? You also seem to have two entries for it. Delete one.

**Comment by Atque** [comment]
@axd1967 The comet symbol turning its tail in the opposite direction of the sun would be really nice. I have no idea how much harder that would be to implement. I had in mind a simple symbol with constant tail direction.

**Comment by axd1967** [comment]
> Alright. Looks like the next comet with obviously false MPC data. Please edit its elements and reduce absolute magnitude by ~20 or 23 magnitudes... Or did I miss something big? You also seem to have two entries for it. Delete one.

yes... I was long time wondering what went wrong, but it's not a big issue. Don't worry, I take it with a lot of salt ;-) in this case it's more to show the current Stellarium behaviour.

(A, B and C are three distinct comets from the MPC sources).

![image](https://user-images.githubusercontent.com/3529789/95350306-50f66480-08c0-11eb-82e4-94d64ded9ae6.png)

![image](https://user-images.githubusercontent.com/3529789/95350535-8dc25b80-08c0-11eb-84cf-a297e436c6e7.png)

**Comment by alex-w** [comment]
@Atque I have question for 3rd item (Being able to use more customizable labels.): Is smart mode unacceptable for you?

**Comment by Atque** [comment]
@alex-w Smart mode is not unacceptable, but I think it could be too limiting.

**Comment by alex-w** [comment]
Please check version 0.20.3-4b8e7fd1c8: https://github.com/Stellarium/stellarium-data/releases/tag/weekly-snapshot

**Comment by alex-w** [comment]
@Atque I have a question for the 1st item: maybe just create markers according to the type of celestial body (a "smart marker")?

**Comment by Atque** [comment]
@alex-w Yes, that could be useful. But in that case, I would like to have two options: 1. Use "smart markers", and 2. Use normal markers (the circle marker used today).

What do you think about the comet marker with a tail always pointing away from the sun? How much more complicated would that be to implement, compared to a simple static comet symbol?

**Comment by axd1967** [comment]
> @alex-w Yes, that could be useful. But in that case, I would like to have two options: 1. Use "smart markers", and 2. Use normal markers (the circle marker used today).
> 
> What do you think about the comet marker with a tail always pointing away from the sun? How much more complicated would that be to implement, compared to a simple static comet symbol?

I find this interesting because the workaround is to use the angle measurement tool to get an idea of the sun angle.

![image](https://user-images.githubusercontent.com/3529789/95835763-ed948880-0d3e-11eb-9522-2558dce0b7f0.png)

the length could be configurable in pixels.
